Code/GraphMol/ChemTransforms/ChemTransforms.cpp

```diff
@@ -347,8 +359,13 @@ ROMol *replaceCore(const ROMol &mol, const ROMol &core,
                    const MatchVectType &matchV, bool replaceDummies,
                    bool labelByIndex, bool requireDummyMatch) {
   unsigned int origNumAtoms = mol.getNumAtoms();
+  std::vector<std::pair<int, SideChainMapping>> matches;
+  matches.reserve(origNumAtoms);
+
   std::vector<int> matchingIndices(origNumAtoms, -1);
   std::vector<int> allIndices(origNumAtoms, -1);
+  boost::dynamic_bitset<> molAtomsMapped(origNumAtoms);
+  std::set<int> multipleMappedMolAtoms;
```

Comment:
It's probably better to use a dynamic_bitset than an std::set for this "flag the interesting items and then loop over items and check if they are interesting" pattern. 
The code ends up being a tiny bit simpler and the bitset alternative has been faster every time I've tested it.

I assume this would end up being different if the potential length of the bitset were huge, but that's not going to be the case with atom indices.

Reply:
Changed